docker-compose.yml

```diff
@@ -0,0 +1,11 @@
+version: '3.8'
+
+services:
+  database:
+    container_name: leftovers-postgres
+    image: postgres:16.3
+    restart: always
+    ports:
+      - 5432:5432
+    env_file:
+      - ".env"
```

Comment:
I recommend that you add the following:

`volumes` - this will create a docker volume where data for the database will be stored. This way when you stop the container / do `docker compose down`, you won't loose the data (you still can loose the data if you remove the docker volume).

`networks` - this way you create a shared network between all services running via the docker compose, and they will be able to see / communicate with each other. Using `driver: bridge` lets the host also see the services exposed by the containers on `localhost`.

```
services:
  database:
    ...
    volumes:
      - postgres-data:/data/postgres
    networks:
      - fullstack-project

...

networks:
  fullstack-project:
    driver: bridge

volumes:
  postgres-data:
```

Reply:
Also, looking at the postgres container documentation, I'm not sure if you don't have to specify this:

```
    environment:
      PGDATA: /data/postgres
```

Maybe not, but worth checking!